clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3040,6 +3040,30 @@ static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D,
   D->addAttr(::new (S.Context) SYCLIntelNumSimdWorkItemsAttr(
         S.Context, Attr, NumSimdWorkItems));
 }
+// Handles scheduler_target_fmax_mhz
+static void handleSchedulerTargetFmaxMhzAttr(Sema &S, Decl *D,
```

Comment:
This function seems so incredibly similar to the one above it, and I cannot imagine we don't have a couple more that are exactly like this.  I think we should use this opportunity to merge the handling into 1 function template.

Reply:
My code changed but it can be refactored for better code reuse. 
Alas, I just noticed this - https://github.com/intel/llvm/pull/2510 ;looks like Soumi and I were trying to do similar things. 
Next patch to follow this.